### PR TITLE
docs: add examples for localize

### PR DESCRIPTION
**Pull request by santoshyadavdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: adev/src/content/guide/i18n/prepare.md [hunk 1]
```diff
@@ -109,6 +109,14 @@ $localize `string_to_translate ${variable_name}`;
 $localize `string_to_translate ${variable_name}:placeholder_name:`;
 </docs-code>
 
+### Conditional syntax for translations
+
+<docs-code language="typescript">
+return this.show ? $localize `Show Tabs` : $localize `Hide tabs`;
```
Comment (JeanMeche) [inline adev/src/content/guide/i18n/prepare.md:115]: I know our docs aren't really consistent on this, but wdyt of removing the trailing spaces after `$localize` ?

Comment (santoshyadavdev) [reply adev/src/content/guide/i18n/prepare.md:115]: yeah, good catch, I will do that; I am considering adding more examples.

Comment (santoshyadavdev) [reply adev/src/content/guide/i18n/prepare.md]: udpated all instances

**Comment by LayZeeDK** [comment]
Maybe this should be a different effort but an example of lazy loading translation texts would do wonders.

File: adev/src/content/examples/i18n/src/app/app.component.html [hunk 2]
```diff
@@ -33,3 +33,8 @@ <h1 i18n="User welcome|An introduction header for this sample@@introductionHeade
   other {{{ minutes }} minutes ago by {gender, select, male {male} female {female} other {other}}}}
 </span>
 <!--#enddocregion i18n-nested-->
+<br><br>
+<!--#docregion i18n-conditional-->
+<button type="button" (click)="toggleDisplay()">Toggle</button>
+<div i18n [attr.aria-label]="toggleStatus" i18n-aria-label>{{toggleStatus}}</div>
```
Comment (IKatsuba) [inline adev/src/content/examples/i18n/src/app/app.component.html:39]: Maybe `i18n-aria-label` is unnecessary here? You already used `$localize` for `toggleStatus`

File: adev/src/content/examples/i18n/src/app/app.component.ts [hunk 3]
```diff
@@ -23,4 +35,7 @@ export class AppComponent {
   other() {
     this.gender = 'other';
   }
+  toggleDisplay() {
+    this.toggle.update((toggle) => !this.toggle);
```
Comment (IKatsuba) [inline adev/src/content/examples/i18n/src/app/app.component.ts:39]:
```suggestion
    this.toggle.update((toggle) => !this.toggle());
```

`this.toggle` is a signal. Do we have to call it to get value?

File: adev/src/content/examples/i18n/src/app/app.component.html (hunk 2 above)
Comment (santoshyadavdev) [reply adev/src/content/examples/i18n/src/app/app.component.html:39]: yeah makes sense

File: adev/src/content/examples/i18n/src/app/app.component.ts (hunk 3 above)
Comment (santoshyadavdev) [reply adev/src/content/examples/i18n/src/app/app.component.ts:39]: Thanks @JeanMeche @IKatsuba

**Comment by santoshyadavdev** [comment]
@JeanMeche this is ready for review, if I come across more examples I will open a new PR.

File: adev/src/content/examples/i18n/src/app/app.component.ts [hunk 4]
```diff
@@ -11,6 +12,17 @@ export class AppComponent {
   gender = 'female';
   fly = true;
   logo = '${this.baseUrl}/angular.svg';
+  toggle = signal(false);
+  toggleStatus = '';
+
+  constructor() {
+    effect(() => {
```
Comment (JeanMeche) [inline adev/src/content/examples/i18n/src/app/app.component.ts:19]: Wdyt of making this a `computed` instead of an `effect` ?

Comment (santoshyadavdev) [reply adev/src/content/examples/i18n/src/app/app.component.ts]: Done updated the code

**Review by JeanMeche — APPROVED** [review]
Thank you Santosh for the docs improvement, it's always welcome !

**Comment by thePunderWoman** [comment]
This PR was merged into the repository by commit bc03f53337c113c619f888d9dee8c7b3b8ff9366.

The changes were merged into the following branches: main, 18.2.x

**Pull request closed without merging**